OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs

```diff
@@ -53,9 +54,17 @@ public GameInfoStatsLogic(Widget widget, World world)
 				nameLabel.GetColor = () => pp.Color.RGB;
 
 				var flag = item.Get<ImageWidget>("FACTIONFLAG");
-				flag.GetImageName = () => pp.Country.Race;
 				flag.GetImageCollection = () => "flags";
-				item.Get<LabelWidget>("FACTION").GetText = () => pp.Country.Name;
+				if (pp.World.LocalPlayer.Stances[pp] == Stance.Ally)
+				{
+					flag.GetImageName = () => pp.Country.Race;
+					item.Get<LabelWidget>("FACTION").GetText = () => pp.Country.Name;
+				}
+				else
+				{
+					flag.GetImageName = () => pp.PlayerReference.Race;
+					item.Get<LabelWidget>("FACTION").GetText = () => "Unknown";
```

Comment:
This should show the proper label for the selected random type (there are now multiple in RA).  "Unknown" would be suitable if they were later discovered, but we aren't doing that in this pr.